File: Find First Palindromic String in the Array.js
// solution 1

/**
 * @param {string[]} words
 * @return {string}
 */
var firstPalindrome = function(words) {
    for (let word of words) {
        if (word === word.split('').reverse().join('')) {
            return word;
        }
    }
    return '';    
};



// solution 2
/**
 * @param {string[]} words
 * @return {string}
 */
var firstPalindrome = function(words) {
    for (let word of words) {
        let left = 0;
        let right = word.length - 1;
        
        while (word[left] === word[right]) {
            left++;
            right--;
            if (left >= right) {
                return word;
            }
        }
    }
    return '';    
};

File: Find First Palindromic String in the Array.cpp

// Solution 1
class Solution {
public:
    string firstPalindrome(vector<string>& words) {
        for (string word : words) {
            if (word == string(word.rbegin(), word.rend())) {
                return word;
            }
        }
        return "";        
    }
};

// Solution 2
class Solution {
public:
    string firstPalindrome(vector<string>& words) {
        for (string word : words) {
            int left = 0;
            int right = word.length() - 1;
            
            while (left < right) {
                if (word[left] != word[right]) {
                    break;
                }
                left++;
                right--;
            }
            if (left >= right) {
                return word;
            }
        }
        return "";        
    }
};
